python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -107,13 +107,17 @@ class TableMetadata:
     column_details: list
     retention_period_in_days: int | None
     domain: str | None = None
-    parent_database_name: str | None = None
+    parent_entity_name: str | None = None
+    relationships: dict | None = None
```

Comment:
Is `relationships` the right word here? I was thinking this is just `parents`? 

If we want this to be able to handle any kind of relationship I think there should be some way of inspecting the relationship type.

Maybe we could introduce another object for this, e.g.

```python

class RelationshipType(Enum):
  PARENT = auto()


class RelatedAsset:
  id: str
  name: str


class TableMetadata:
  relationships: dict[RelationshipType, list[RelatedAsset]]
```

`parent_entity_name` would then be redundant, and in the frontend we could iterate over any kind of relationship and render links to each one

Reply:
totally agree. Although `parent_entity_name` is also used for ingestion purposes, so i'll still keep that